cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -302,19 +299,38 @@ void AddSortingKernels(VectorKernel base, VectorFunction* func) {
   }
 }
 
+FunctionDoc sort_indices_doc(
+    "Return the indices that would sort an array",
+    ("Compute an array of indices that define a non-stable sort\n"
+     "of the input array.  Null values are considered greater than any\n"
+     "other value and are therefore sorted at the end of the array."),
+    {"array"});
+
+FunctionDoc partition_nth_indices_doc(
+    "Return the indices that would partition an array around a pivot",
+    ("Compute an array of indices that define a non-stable partition\n"
+     "around the `N`th smallest element of the input array.\n"
```

Comment:
For me it's not fully clear what this "partition around the Nth smallest element" means.

Reply:
I think what you wrote in the compute.rst guide is clearer:

> * \(1) The output is an array of indices into the input array, that define
  a partial sort such that the *N*'th index points to the *N*'th element
  in sorted order, and all indices before the *N*'th point to elements
  less or equal to elements at or after the *N*'th (similar to
  :func:`std::nth_element`).  *N* is given in
  :member:`PartitionNthOptions::pivot`.

Reply:
Will improve this, thank you.